examples/OpenTracing.Examples/CommonRequestHandler/Client.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace OpenTracing.Examples.CommonRequestHandler
+{
+    public class Client
+    {
+        private readonly RequestHandler _requestHandler;
+
+        public Client(RequestHandler requestHandler)
+        {
+            _requestHandler = requestHandler;
+        }
+
+        public async Task<String> Send(String message)
+        {
```

Comment:
This code is a bit weird IMO and looks very non-C#y :smile:

Wouldn't the Java code translate to a very simple async/await pattern like the following?

```csharp
var context = new Context();

await Task.Delay(50);
_requestHandler.BeforeRequest(message, context);
await Task.Delay(50);
_requestHandler.AfterResponse(message, context);
return $"{message}:response";
```